### PR TITLE
Most recently viewed reviews

### PR DESCRIPTION
**Pull request by x3igh7**
Not quite ready for merge, but available for code review!

### CODE REVIEW

File: app/assets/stylesheets/home.css.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  margin-bottom: 20px;
+}
+
+.hero-unit {
+  width: 100%;
+}
+
+.sign-up-link {
+  padding-top: 35px;
+}
```
Comment (jzgit) [inline app/assets/stylesheets/home.css.scss:14]: newline!

File: app/models/recent_review.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class RecentReview < ActiveRecord::Base
+  attr_accessible :user_id, :review_id
```
Comment (jzgit) [inline app/models/recent_review.rb]: let's use the actual objects here

```
attr_accessible :user, :review
```

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -21,4 +21,12 @@ def index
     @movie = Movie.find(params[:movie_id])
     @reviews = @movie.reviews
   end
+
+  def show
+    @user = current_user
+    @movie = Movie.find(params[:movie_id])
+    @review = Review.find(params[:id])
+
+    RecentReview.create(:user_id => @user.id, :review_id => @review.id)
```
Comment (jzgit) [inline app/controllers/reviews_controller.rb]: use the actual objects here

```
RecentReview.create(:user => @user, :review => @review)
```

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class UsersController < ApplicationController
+
+  def show
+    @user = User.find(params[:id])
+    @recent_review = RecentReview.where("user_id = #{@user.id}")
```
Comment (jzgit) [inline app/controllers/users_controller.rb]: dont need to create an extra ivar here if you use it on the next line

this should work:

```
@reviews = RecentReview.where("user_id = ?", user.id).limit(3)
```

File: app/views/reviews/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<h1><%= @review.title %></h1>
+<h3><%= @review.body %></h3>
```
Comment (jzgit) [inline app/views/reviews/show.html.erb:2]: newline!

File: app/views/users/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<p>Profile for <%= @user.email %></p>
+
+<div>
+  <% @reviews.each do |f| %>
+    <p><%= Review.find(f.review_id).title %></p>
```
Comment (jzgit) [inline app/views/users/show.html.erb]: @reviews is an array of RecentReviews here so you should be able to get to the review associated with each inside your loop (if you set up your associations correctly):

```
f.review.title
```

File: app/models/recent_review.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+
+  has_many :users
```
Comment (jzgit) [inline app/models/recent_review.rb]: i think you may have your associations backwards - a review has_many recent_reviews

a recent_review belongs_to :user
a recent_review belongs_to :review

**Comment by x3igh7** [comment]
It's... done?...

File: app/controllers/reviews_controller.rb [hunk 8]
```diff
@@ -22,4 +20,22 @@ def index
...
     @like = @movie.likes.new
   end
-end
\ No newline at end of file
+
+  def show
+    @user = current_user
+    @movie = Movie.find(params[:movie_id])
+    @review = Review.find(params[:id])
+    @recent_reviews = RecentReview.where("user_id = ?", @user.id).limit(3)
```
Comment (x3igh7) [inline app/controllers/reviews_controller.rb:28]: Jason, should we use last instead of limit here?

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+class UsersController < ApplicationController
+
+  def show
+    @user = User.find(params[:id])
+    @reviews = RecentReview.where("user_id = ?", @user.id).limit(3)
```
Comment (x3igh7) [inline app/controllers/users_controller.rb:5]: Same question here: 

Should we use last instead of limit?

**Pull request closed without merging**